File: sdk/agentred/resources/targets.py
"""Targets API resource."""
from typing import Optional


class TargetsResource:
    """Manage target configurations."""

    def __init__(self, client):
        self._c = client

    def create(
        self,
        name: str,
        target_type: str,
        endpoint: str,
        api_key: Optional[str] = None,
        **kwargs
    ) -> dict:
        """Create a new target."""
        payload = {
            "name": name,
            "target_type": target_type,
            "endpoint": endpoint,
            **kwargs
        }
        if api_key:
            payload["api_key"] = api_key
        return self._c._request("POST", "/targets", json=payload)

    def get(self, target_id: str) -> dict:
        """Get target by ID."""
        return self._c._request("GET", f"/targets/{target_id}")

    def list(self) -> list:
        """List all targets."""
        result = self._c._request("GET", "/targets")
        return result.get("items", []) if isinstance(result, dict) else result

    def delete(self, target_id: str) -> dict:
        """Delete a target."""
        return self._c._request("DELETE", f"/targets/{target_id}")

    def test_connection(self, target_id: str) -> dict:
        """Test connection to target."""
        return self._c._request("POST", f"/targets/{target_id}/test", json={})

    def update(self, target_id: str, **kwargs) -> dict:
        """Update target configuration."""
        return self._c._request("PATCH", f"/targets/{target_id}", json=kwargs)

File: sdk/js/src/resources/targets.ts
import { AxiosInstance } from 'axios';
import { Target, CreateTargetRequest } from '../types';

export class TargetsResource {
  constructor(private http: AxiosInstance) {}

  async list(): Promise<Target[]> {
    const { data } = await this.http.get('/targets');
    return data;
  }

  async create(request: CreateTargetRequest): Promise<Target> {
    const { data } = await this.http.post('/targets', request);
    return data;
  }

  async get(targetId: string): Promise<Target> {
    const { data } = await this.http.get(`/targets/${targetId}`);
    return data;
  }

  async update(targetId: string, updates: Partial<CreateTargetRequest>): Promise<Target> {
    const { data } = await this.http.patch(`/targets/${targetId}`, updates);
    return data;
  }

  async delete(targetId: string): Promise<void> {
    await this.http.delete(`/targets/${targetId}`);
  }

  async testConnection(targetId: string): Promise<{ connected: boolean; latency_ms: number; error?: string }> {
    const { data } = await this.http.post(`/targets/${targetId}/test`);
    return data;
  }
}
